tests/components/intellifire/conftest.py

```diff
@@ -17,35 +41,241 @@ def mock_setup_entry() -> Generator[AsyncMock]:
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireCloudInterface.user_data",
+        new_callable=PropertyMock,
+    ) as mock_user_data:
+        mock_user_data.return_value = user_data_mock
+        yield
+
+
+@pytest.fixture
+def mock_cloud_api_interface_user_data_3() -> IntelliFireUserData:
+    """Fixture to mock the user_data property of IntelliFireCloudInterface."""
+    fixture_data = load_fixture("intellifire/user_data_3.json")
+    user_data_mock = IntelliFireUserData(**json.loads(fixture_data))
+
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireCloudInterface.user_data",
+        new_callable=PropertyMock,
+    ) as mock_user_data:
+        mock_user_data.return_value = user_data_mock
+        yield
+
+
+@pytest.fixture
+def mock_local_poll_data() -> IntelliFirePollData:
+    """Mock polling a local fireplace."""
+    fixture_data = load_fixture("intellifire/local_poll.json")
+    poll_data_mock = IntelliFirePollData(**json.loads(fixture_data))
+
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireAPILocal.data",
+        new_callable=PropertyMock,
+    ) as mock_user_data:
+        mock_user_data.return_value = poll_data_mock
+        yield
+
+
+@pytest.fixture
+def mock_login_with_credentials() -> Generator[AsyncMock, None, None]:
+    """Mock the login_with_credentials method and set _user_data."""
+
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireCloudInterface.login_with_credentials",
+        new_callable=AsyncMock,
+    ) as mock_login:
+        # mock_login.side_effect = login_side_effect
+        yield mock_login
+
+
+@pytest.fixture
+def mock_login_with_bad_credentials() -> Generator[AsyncMock, None, None]:
+    """Mock the login_with_credentials method and set _user_data."""
+
+    # Custom side effect function
+    async def side_effect_func(*args, **kwargs):
+        if not hasattr(side_effect_func, "called"):
+            side_effect_func.called = True
+            raise LoginError
+        # Simulate async function
+
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireCloudInterface.login_with_credentials",
+        new_callable=AsyncMock,
+    ) as mock_login:
+        mock_login.side_effect = side_effect_func
+        yield mock_login
+
+
+@pytest.fixture
+def mock_poll_local_fireplace() -> Generator[AsyncMock, None, None]:
+    """Mock polling of local fireplace."""
+    with patch(
+        "homeassistant.components.intellifire.config_flow.IntelliFireAPILocal.poll",
+        new_callable=AsyncMock,
+    ) as mock_poll:
+        # mock_login.side_effect = login_side_effect
+        yield mock_poll
+
+
+@pytest.fixture
+def mock_cloud_poll() -> Generator[AsyncMock, None, None]:
+    """Mock a successful cloud poll call."""
+    with patch(
+        "homeassistant.components.intellifire.__init__.UnifiedFireplace.perform_cloud_poll",
```

Comment:
```suggestion
        "homeassistant.components.intellifire.UnifiedFireplace.perform_cloud_poll",
```
Init is the same as the root of the project